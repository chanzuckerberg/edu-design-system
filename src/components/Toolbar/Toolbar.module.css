/*------------------------------------*\
    # TOOLBAR
\*------------------------------------*/

/**
 * A toolbar is a container that houses form controls for uses things like filtering a data table.
 */
.toolbar {
  display: flex;
  align-items: center;
  gap: var(--eds-size-2);
  padding-top: var(--eds-size-1);
  padding-right: var(--eds-size-2);
  padding-bottom: var(--eds-size-1);
  padding-left: var(--eds-size-2);
  background-color: var(--eds-theme-color-background-neutral-subtle);
}

/**
 * Stripped-down toolbar without background color or padding 
 */
.toolbar--bare {
  background-color: transparent;
  padding: 0;
}

/**
 * Veritcally aligns toolbar items to the top of the toolbar.
 */
.toolbar--vertical-align-top {
  align-items: flex-start;
}

/**
 * Veritcally aligns toolbar items to the bottom of the toolbar.
 */
.toolbar--vertical-align-bottom {
  align-items: flex-end;
}

/**
 * Center-aligned toolbar item
 */
.toolbar__item--align-center {
  /**
   * Since the toolbar is set to display: flex, using margin: auto will
   * push item away from other items.
   */
  margin-left: auto;
  margin-right: auto;
}

/**
 * Right-aligned toolbar item
 */
.toolbar__item--align-right {
  /**
   * Since the toolbar is set to display: flex, using margin: auto will
   * push item away from other items.
   */
  margin-left: auto;
}
