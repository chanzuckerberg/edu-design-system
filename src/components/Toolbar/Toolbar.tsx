import React, { ReactNode } from 'react';
import clsx from 'clsx';
import styles from './Toolbar.module.css';

export interface Props {
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Vertical align
   * - **bottom** vertically aligns toolbar items to the bottom of toolbar
   * - **top** vertically aligns toolbar items to the top of toolbar
   */
  variant?: 'bare';
  /**
   * Vertical align:
   * - **bottom** vertically aligns t
   */
  verticalAlign?: 'bottom' | 'top';
}

/**
 * Primary UI component for user interaction
 */
export const Toolbar: React.FC<Props> = ({
  className,
  children,
  variant,
  verticalAlign,
  ...other
}) => {
  const componentClassName = clsx(styles['toolbar'], className, {
    [styles['toolbar--bare']]: variant === 'bare',
    [styles['toolbar--vertical-align-bottom']]: verticalAlign === 'bottom',
    [styles['toolbar--vertical-align-top']]: verticalAlign === 'top',
  });
  return (
    <div className={componentClassName} {...other}>
      {children}
    </div>
  );
};
