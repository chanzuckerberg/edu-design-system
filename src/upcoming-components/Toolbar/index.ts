export { Toolbar as default } from './Toolbar';
