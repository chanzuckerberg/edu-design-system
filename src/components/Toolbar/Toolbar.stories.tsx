import type {StoryObj, Meta} from '@storybook/react';
import React from 'react';

import {Toolbar} from './Toolbar';
import ToolbarItem from '../ToolbarItem';

export default {
  title: 'Organisms/Toolbars/Toolbar',
  component: Toolbar,
  subcomponents: {ToolbarItem},
  parameters: {
    axe: {
      // TODO: re-enable when component is worked on
      skip: true,
    },
  },
  args: {
    children: (
      <>
        <Toolbar.Item>
          <div className="fpo u-margin-none">Left</div>
        </Toolbar.Item>
        <Toolbar.Item>
          <div className="fpo u-margin-none">Left</div>
        </Toolbar.Item>
        <Toolbar.Item>
          <div className="fpo u-margin-none">Left</div>
        </Toolbar.Item>
        <Toolbar.Item align="right">
          <div className="fpo u-margin-none">Right</div>
        </Toolbar.Item>
      </>
    ),
  },
} as Meta<Args>;

type Args = React.ComponentProps<typeof Toolbar>;

export const Default: StoryObj<Args> = {};
export const Bare: StoryObj<Args> = {
  args: {
    variant: 'bare',
  },
};

export const AlignTop: StoryObj<Args> = {
  args: {
    verticalAlign: 'top',
  },
};

export const AlignBottom: StoryObj<Args> = {
  args: {
    verticalAlign: 'bottom',
  },
};
