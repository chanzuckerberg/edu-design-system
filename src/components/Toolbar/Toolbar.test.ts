import {generateSnapshots} from '@chanzuckerberg/story-utils';
import * as stories from './Toolbar.stories';

describe('<Toolbar />', () => {
  generateSnapshots(stories);
});
