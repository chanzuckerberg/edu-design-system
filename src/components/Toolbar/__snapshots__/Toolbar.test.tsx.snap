// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> AlignBottom story renders snapshot 1`] = `
<div
  class="toolbar toolbar--vertical-align-bottom"
>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item toolbar__item--align-right"
  >
    <div
      class="fpo u-margin-none"
    >
      Right
    </div>
  </div>
</div>
`;

exports[`<Toolbar /> AlignTop story renders snapshot 1`] = `
<div
  class="toolbar toolbar--vertical-align-top"
>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item toolbar__item--align-right"
  >
    <div
      class="fpo u-margin-none"
    >
      Right
    </div>
  </div>
</div>
`;

exports[`<Toolbar /> Bare story renders snapshot 1`] = `
<div
  class="toolbar toolbar--bare"
>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item toolbar__item--align-right"
  >
    <div
      class="fpo u-margin-none"
    >
      Right
    </div>
  </div>
</div>
`;

exports[`<Toolbar /> Default story renders snapshot 1`] = `
<div
  class="toolbar"
>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item"
  >
    <div
      class="fpo u-margin-none"
    >
      Left
    </div>
  </div>
  <div
    class="toolbar__item toolbar__item--align-right"
  >
    <div
      class="fpo u-margin-none"
    >
      Right
    </div>
  </div>
</div>
`;
